crates/api_models/src/admin.rs

```diff
@@ -477,6 +477,51 @@ pub struct MerchantConnectorCreate {
 
     #[schema(value_type = Option<ConnectorStatus>, example = "inactive")]
     pub status: Option<api_enums::ConnectorStatus>,
+
+    /// In case the merchant needs to store any additional sensitive data
+    #[schema(value_type = Option<AdditionalMerchantData>)]
+    pub additional_merchant_data: Option<pii::SecretSerdeValue>,
```

Comment:
expecting it to have multiple variants with with complex nested structs underneath, hence thought of keeping it as a json.

Reply:
If it's in the API, it's better for it to be a concrete type since any structural validations will be handled by the API layer itself and you won't have to manually parse it and throw errors.